### PR TITLE
update check

### LINKED ISSUE #762: feat: update checker
We should have messaging when we have new versions, since we plan on having a regular cadence. Nothing fancy, something like https://www.npmjs.com/package/update-check should do the trick.

### PR DESCRIPTION
**Pull request by JacobMGEvans**
feat: Added the update check that will check the package once a day against the beta release, `distTag` can be changed later, then prints the latest beta version to the user.

resolves #762

### CODE REVIEW

File: packages/wrangler/src/cli.ts [hunk 1]
```diff
@@ -2,10 +2,12 @@ import process from "process";
 import { hideBin } from "yargs/helpers";
 import { FatalError } from "./errors";
 import { reportError, initReporting } from "./reporting";
+import { updateCheck } from "./update-check";
 import { main } from ".";
 
 try {
   initReporting();
+  void updateCheck();
```
Comment (JacobMGEvans) [inline packages/wrangler/src/cli.ts:10]: This is probably redundant because it has a `try/catch` in the `updateCheck`

**Comment by JacobMGEvans** [comment]
Alternatively, I started using Ink to improve the UI of this output. I think if we do that it would need to be called around the same place as `yargs.parse()`

**Comment by JacobMGEvans** [comment]
Screen Shot 2022-04-05 at 1 31 05 PM

**Comment by JacobMGEvans** [comment]
This is with the Ink, implementation I spent some time on. 

Screen Shot 2022-04-05 at 2 56 32 PM

File: packages/wrangler/src/update-check.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+export async function updateCheck(): Promise<JSX.Element | null> {
+  let update = null;
+  try {
+    // default cache for update check is 1 day
+    update = await checkForUpdate(pkg, {
+      distTag: "beta",
+    });
+  } catch (err) {
+    console.error(`Failed to check for updates: ${err}`);
```
Comment (threepointone) [inline packages/wrangler/src/update-check.ts:13]: This should probably be a silent failure

File: packages/wrangler/src/update-check.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  }
+
+  if (update)
```
Comment (threepointone) [inline packages/wrangler/src/update-check.ts:16]: Does this mean we'll always show this message if there's an update? Do you suppose we only want to show this message once a day?

**Review by threepointone — COMMENTED** [review]
Let's not use ink for this just yet.

**Comment by threepointone** [comment]
Another idea is to include inside `printWranglerBanner()`, so it's less aggressive. `⛅️ wrangler 0.0.24 (update available 0.0.25)`. What do you think?

File: packages/wrangler/src/update-check.ts (hunk 3 above)
Comment (JacobMGEvans) [reply packages/wrangler/src/update-check.ts:16]: The default is it checks once a day.

**Comment by JacobMGEvans** [comment]
> Another idea is to include inside `printWranglerBanner()`, so it's less aggressive. `⛅️ wrangler 0.0.24 (update available 0.0.25)`. What do you think?

I think it is a good idea, we can always pivot later if we want something more like what NPM or similar packages do for their update messaging.

**Pull request merged**